bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -275,32 +275,37 @@ public static Set<Type> getConvertibleTypes(Object inputValue, Type targetType,
                     if ((inputValueType == memType) || isIntegerSubtypeAndConvertible(inputValue, memType)) {
```

Comment:
I also think rather than doing what is done here and in https://github.com/HindujaB/ballerina-lang/pull/1, because `getConvertibleTypes()` will return all convertible types, and `isConvertible.. ()` methods will return whether they are convertible, rather than tampering with them, need to introduce a proper method here in this if condition which will check whether the given inputValue belongs to the the `memType`

Reply:
If we are to do that, then that implementation needs to check all the internal elements of the respected type (Eg. array elements, record fields, etc.). As we already have those implementations in `getConvertibleTypes()`,  IMO, it is better to add these validations inside those methods.